src/connectors/range/connectRange.js

```diff
@@ -292,6 +292,59 @@ export default function connectRange(renderFn, unmountFn) {
 
         return nextState;
       },
+
+      getWidgetState(fullState, { state }) {
```

Comment:
Same comments as for `priceRanges` since it's exactly the same logic. Maybe we could refactor a bit this part to reuse the same function across both widgets.

Reply:
I'm not in favor to merge part of the logic of two different connectors. This would mean that we would have a third party holding the logic of a connector or even multiple connectors. I have the impression that  would make the code harder to follow.

However this might hint for a generalization of ranges here, right? In this case, we could have only one generalized connector. What about digging into that later?

Reply:
Completely agree, we can dig into that later.